agent/structs/intention.go

```diff
@@ -325,11 +322,12 @@ func (ixn *Intention) CanRead(authz acl.Authorizer) bool {
 }
 
 func (ixn *Intention) CanWrite(authz acl.Authorizer) bool {
-	if authz == nil || authz == acl.ManageAll() {
+	if authz == acl.ManageAll() {
 		return true
 	}
 	var authzContext acl.AuthorizerContext
 
+	// TODO: this line seems to require checking 'authz == acl.ManageAll()' above
 	if ixn.DestinationName == "" {
 		return false
 	}
```

Comment:
Calling out this TODO. In all other cases we've found (so far) simply using the `acl.Authorizer` without checking its value is correct.

In this case it seems like it may not be , so we had to add an `authz == acl.ManangeAll()` check above (the `ManageAll` authorized is returned when acls are disabled).

Is this `ixn.DestinationName == ""` in the right place? Maybe it should be in `authz.IntentionWrite` (called below)?   I believe if we made that change we could remove the `authz == acl.ManangeAll()` the `acl.Authorizer` interface would be much safer to use in general.

Reply:
cc @freddygv @rboyer  Thoughts on this proposed change? Any concerns?

Reply:
Sounds good. I'll address this in a follow up as well